coreos-installer

```diff
@@ -520,6 +520,14 @@ download_image() {
         echo "${PCT}%" >&2
         sleep 1
     done
+
+    wait $curlpid
```

Comment:
Aside, though the `ps --pid` trick isn't foolproof since a new process might take up that same pid.

Offhand another approach could be: `(curl ...; echo $? > /tmp/curl-rc) &`. Then the loop can be `while [ ! -f /tmp/curl-rc ]` and we can drop the `curlpid` var and the `wait` too.

Reply:
hmm - never used  `wait` before. Does it work for processes that have already completed?

Reply:
Yeah, it'll just return immediately. Note it only works with child processes of the shell though (see `help wait`).

Reply:
welp - learned something new today

Reply:
I liked the echo RC to file approach and have pushed a change switching to it.